Added new selectable gradient + accent colors

marcelweikum opened the pull request:
Hey Victor,

I now added the feature which I requested earlier. https://github.com/victorbalssa/abacus/issues/319#issue-2413149510

Main changes are:

- New Color Selection Screen in the Configuration Screen:
  - Here you can select a color that is saved to redux storage. Even after a restart the selected color is still selected.

- The gradients are now changed into a compacted version, so you only have to declare them in colors.ts

- I added translations for all available languages for my feature

- Gradients are linked to Accent Color, when I change to a color, the gradient on the Home Screen as well as the colors for the Buttons, Graphs (only first graph, the others are random) and Switches change too.

- I changed the TransactionForm Input Background to a more transparent version of the selected Color (That technically is the same as before but now with different colors)

<img src="https://github.com/user-attachments/assets/6b3819d0-86a3-4699-b5ad-ae091da50f56" alt="Simulator Screenshot - iPhone 15 - 2024-07-17 at 21 46 45" width="300" />

<img src="https://github.com/user-attachments/assets/ca707bd4-e989-404d-ada4-348fc0016b09" alt="Simulator Screenshot - iPhone 15 - 2024-07-17 at 21 46 28" width="300" />

<img src="https://github.com/user-attachments/assets/164598ac-462e-48a3-84c8-6a5d399bc7a2" alt="Simulator Screenshot - iPhone 15 - 2024-07-17 at 21 47 08" width="300" />

If there are any suggestions or bugs, do not hesitate to contact me!

Best regards,
Marcel

marcelweikum committed:
Changed HomeScreen gradient to variables instead of hardcoded

marcelweikum committed:
Added translations for eng and ger

marcelweikum committed:
Added translations for the new color feature

marcelweikum committed:
Added Tab for customization and changed colors

marcelweikum committed:
Added variable color

marcelweikum committed:
Added gradient for dark and light

marcelweikum committed:
Changed primary to be selected color

marcelweikum committed:
Added variable color and changed background to a more transparent version of the selected color

marcelweikum committed:
Added variable color

marcelweikum committed:
Added new colors and gradients

marcelweikum committed:
Added theme selection functions

marcelweikum committed:
Changed the graph color so the first Graph is the selected color and the others are random

marcelweikum committed:
Added route to change color

marcelweikum committed:
Added Color Selection View

victorbalssa commented on src/components/Screens/HomeScreen.tsx:
```diff
@@ -538,6 +539,11 @@ export default function HomeScreen() {
   const end = useSelector((state: RootState) => state.firefly.rangeDetails.end);
   const currentCode = useSelector((state: RootState) => state.currencies.currentCode);
   const dispatch = useDispatch<RootDispatch>();
+  const selectedTheme = useSelector((state: RootState) => state.configuration.selectedTheme);
```
When user update the app via OTA new store key is `undefined` so we need a way to default to `gradientOrange` until the user select another theme:
 ```suggestion
  const selectedTheme = useSelector((state: RootState) => state.configuration.selectedTheme || 'gradientOrange');```

marcelweikum replied:
True! Updated it. 55b5dc9f0af99d1f85951d3704744a830f4324ee a22bf175aec8ba26e175b450add53bc58bd3e3dc

victorbalssa commented on src/components/UI/ALibrary/AButton.tsx:
```diff
@@ -29,7 +31,8 @@ export default function AButton({
 }: AButtonType) {
   const { colors } = useThemeColors();
-  const backgroundColor = colors[type];
+  const selectedBrandStyle = useSelector((state: RootState) => state.configuration.selectedBrandStyle);
```
same comment here we need to default to `colors.brandStyleOrange`

marcelweikum replied:
Updated it. 55b5dc9f0af99d1f85951d3704744a830f4324ee a22bf175aec8ba26e175b450add53bc58bd3e3dc

victorbalssa commented on src/components/UI/ALibrary/AInput.tsx:
```diff
@@ -52,19 +54,28 @@ export default function AInput({
 }: AInputType) {
   const { colors } = useThemeColors();
   const [isFocused, setIsFocused] = React.useState(false);
+  const selectedBrandStyle = useSelector((state: RootState) => state.configuration.selectedBrandStyle);
```
same comment here we need to default to `colors.brandStyleOrange`

marcelweikum replied:
Updated it. 55b5dc9f0af99d1f85951d3704744a830f4324ee a22bf175aec8ba26e175b450add53bc58bd3e3dc

victorbalssa commented on src/components/UI/Filters.tsx:
```diff
@@ -22,6 +22,7 @@ export default function Filters() {
   const range = useSelector((state: RootState) => state.firefly.rangeDetails.range);
   const accounts = useSelector((state: RootState) => state.accounts.accounts);
   const selectedAccountIds = useSelector((state: RootState) => state.accounts.selectedAccountIds);
+  const selectedBrandStyle = useSelector((state: RootState) => state.configuration.selectedBrandStyle);
```
same comment here we need to default to `colors.brandStyleOrange`

marcelweikum replied:
Updated it. 55b5dc9f0af99d1f85951d3704744a830f4324ee a22bf175aec8ba26e175b450add53bc58bd3e3dc

victorbalssa commented on src/components/Screens/ConfigurationScreen.tsx:
```diff
@@ -37,6 +37,7 @@ export default function ConfigurationScreen({ navigation }: ScreenType) {
   const backendURL = useSelector((state: RootState) => state.configuration.backendURL);
   const useBiometricAuth = useSelector((state: RootState) => state.configuration.useBiometricAuth);
   const dispatch = useDispatch<RootDispatch>();
+  const selectedBrandStyle = useSelector((state: RootState) => state.configuration.selectedBrandStyle);
```
same comment here we need to default to `colors.brandStyleOrange`

marcelweikum replied:
Updated it. 55b5dc9f0af99d1f85951d3704744a830f4324ee a22bf175aec8ba26e175b450add53bc58bd3e3dc

victorbalssa commented on src/components/Screens/ColorSelectionScreen.tsx:
```diff
@@ -0,0 +1,101 @@
+
+export default function ColorSelectionScreen() {
+  const { colors: themeColors } = useThemeColors();
+  const currentTheme = useSelector((state: RootState) => state.configuration.selectedTheme);
```
same comment here we need to default to `gradientOrange`

marcelweikum replied:
Updated it. 55b5dc9f0af99d1f85951d3704744a830f4324ee a22bf175aec8ba26e175b450add53bc58bd3e3dc

victorbalssa commented on src/components/Screens/FilterScreen.tsx:
```diff
@@ -10,8 +10,10 @@ import { useThemeColors } from '../../lib/common';
+
 export default function FilterScreen({ navigation, route }: ScreenType) {
   const { colors } = useThemeColors();
+  const selectedBrandStyle = useSelector((state: RootState) => state.configuration.selectedBrandStyle);
```
```suggestion
  const selectedBrandStyle = useSelector((state: RootState) => state.configuration.selectedBrandStyle || colors.brandStyleOrange);
```

marcelweikum replied:
Updated it. 55b5dc9f0af99d1f85951d3704744a830f4324ee a22bf175aec8ba26e175b450add53bc58bd3e3dc

victorbalssa commented on src/components/Screens/HomeScreen.tsx:
```diff
@@ -44,6 +44,7 @@ function AssetsAccounts() {
   const displayAllAccounts = useSelector((state: RootState) => state.configuration.displayAllAccounts);
   const loading = useSelector((state: RootState) => state.loading.effects.accounts.getAccounts?.loading);
   const dispatch = useDispatch<RootDispatch>();
+  const selectedBrandStyle = useSelector((state: RootState) => state.configuration.selectedBrandStyle);
```
```suggestion
  const selectedBrandStyle = useSelector((state: RootState) => state.configuration.selectedBrandStyle || colors.brandStyleOrange);
```

marcelweikum replied:
Updated it. 55b5dc9f0af99d1f85951d3704744a830f4324ee a22bf175aec8ba26e175b450add53bc58bd3e3dc

victorbalssa commented on src/routes/index.tsx:
```diff
@@ -77,12 +79,13 @@ const styles = StyleSheet.create({
 function TabBarPrimaryButton() {
   const navigation = useNavigation();
   const { colors } = useThemeColors();
+  const selectedBrandStyle = useSelector((state: RootState) => state.configuration.selectedBrandStyle);
```
```suggestion
  const selectedBrandStyle = useSelector((state: RootState) => state.configuration.selectedBrandStyle || colors.brandStyleOrange);
```

marcelweikum replied:
Updated it. 55b5dc9f0af99d1f85951d3704744a830f4324ee a22bf175aec8ba26e175b450add53bc58bd3e3dc

victorbalssa reviewed (commented):
Thanks @marcelweikum, review comments follow.

marcelweikum committed:
Added default value for undefined keys

marcelweikum committed:
Added default to light and dark values

marcelweikum committed:
Changed colors and ColorSelection to fit a more spectrum like UI

marcelweikum committed:
Changed highlighting for selected Color

marcelweikum commented:
I've now added the initial color/gradient for the missing init key.

Also I've changed the ColorSelection to now fit more to the surrounding colors and removed the ScrollView.

<img src="https://github.com/user-attachments/assets/e2e2616d-2dba-4bcf-aefe-648578bc9796" alt="Simulator Screenshot Dark" width="300" />
<img src="https://github.com/user-attachments/assets/7db66fc4-52ef-4a9a-b1af-531403d62443" alt="Simulator Screenshot Light" width="300" />

Another fix I made is the highlighting of the selected color, this was black all the time, no matter if dark mode or light mode is set. Now it sets depending on the mode.

marcelweikum committed:
Added missing color selection to OAuthForm

marcelweikum commented:
Hey again,

I found a missing color Adjustment in efa9c2106c85ad5891519978dbf2dc278c4f7547 and fixed it.

victorbalssa reviewed (approved):
Thank you @marcelweikum, approved.

I will include this work in the next release.
